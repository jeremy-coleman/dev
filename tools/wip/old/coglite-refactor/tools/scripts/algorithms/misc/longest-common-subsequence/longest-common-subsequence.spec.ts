import { lcsLengths, findLCS } from './longest-common-subsequence';

describe('lcsLengths', () => {
  it('should return the table of common subsequences lengths', () => {
    const seqA = 'ABCBDAB';
    const seqB = 'BDCABA';
    const lengths = lcsLengths(seqA, seqB);

    // Add sequences to the matrix for prettier snapshot
    (lengths as any).y = ' ' + seqA;
    (lengths as any).x = ' ' + seqB;

    expect(lengths).toMatchSnapshot('LCS subsequences lengths');
  });
});

describe('findLCS', () => {
  it('should return the LCS for two sequences', () => {
    const seqA = 'ABCBDAB';
    const seqB = 'BDCABA';
    const lcs = findLCS(seqA, seqB);

    expect(lcs).toBe('BCBA');
  });
});
