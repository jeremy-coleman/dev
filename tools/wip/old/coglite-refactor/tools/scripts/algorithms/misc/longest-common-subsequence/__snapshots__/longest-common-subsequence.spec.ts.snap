// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LCS subsequences lengths 1`] = `
     0 1 2 3 4 5 6 
       B D C A B A 
     ______________
0  | 0 0 0 0 0 0 0 
1 A| 0 0 0 0 1 1 1 
2 B| 0 1 1 1 1 2 2 
3 C| 0 1 1 2 2 2 2 
4 B| 0 1 1 2 2 3 3 
5 D| 0 1 2 2 2 3 3 
6 A| 0 1 2 2 3 3 4 
7 B| 0 1 2 2 3 4 4 

`;
